Mage/src/main/java/mage/abilities/AbilityImpl.java

```diff
@@ -559,6 +560,36 @@ public int handleManaXMultiplier(Game game, int value) {
         return xEvent.getAmount();
     }
 
+    /**
+     * Handles X mana costs where each mana has to be a different color (Emblazoned Golem)
+     */
+    private String handleManaXDifferentCost(Game game, Player controller, int amountMana) {
+        if (amountMana >= 5) {
+            return("{W}{U}{B}{R}{G}");
+        }
```

Comment:
If amountMana > 5 I think you should raise an error or something. Something has gone wrong if the player is being asked to pay for more than 5 different colors.

But I am not certain of how this should be handled.